src/path.jl

```diff
@@ -810,17 +806,24 @@ function Base.mktemp(fn::Function, parent::AbstractPath)
     end
 end
 
-function Base.mktempdir(fn::Function, parent::AbstractPath)
-    tmpdir = mktmpdir(parent)
+function Base.mktempdir(fn::Function, parent::AbstractPath; kwargs...)
+    tmpdir = mktmpdir(parent; kwargs...)
     try
         fn(tmpdir)
     finally
         rm(tmpdir, recursive=true)
     end
 end
 
-mktmp(arg1, args...) = mktemp(arg1, args...)
-mktmpdir(arg1, args...) = mktempdir(arg1, args...)
+mktmp(arg1, args...; kwargs...) = mktemp(arg1, args...; kwargs...)
+mktmpdir(arg1, args...; kwargs...) = mktempdir(arg1, args...; kwargs...)
+
+function temp_cleanup(fp::AbstractPath)
+    atexit() do
+        # Might not work in all cases, but default to recursively deleting the path on exit
+        rm(fp; force=true, recursive=true)
```

Comment:
This was the specific suggestion made in #141. It might not always work, but it seems better nothing?